public/app/percona/shared/components/PerconaBootstrapper/PerconaBootstrapper.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { setSettings, setSettingsLoading, setAuthorized } from 'app/percona/shared/core/reducers';
+import { SettingsService } from 'app/percona/settings/Settings.service';
+
+// This component is only responsible for populating the store with Percona's settings initially
+export const PerconaBootstrapper = () => {
```

Comment:
If there's no rendering happening maybe it's better to make this a custom hook (since we are relying on useEffect otherwise it could even be a function). What do you think?

Reply:
Yeah, I also thought of that. The thing is, where should we call the hook? Because there's no "central" component to do it :S

Reply:
In the same component PerconaBootstrapper is being used?

Reply:
Well, I'm putting it on `AppWrapper`, next to the other main components. Two "problems" with it:
- It's not a functional component, so can't use a hook;
- It's where the `<Provider store={store}`> is rendered, so we can't dispatch at this level.

```
<Provider store={store}>
        <ErrorBoundaryAlert style="page">
          <ConfigContext.Provider value={config}>
            <ThemeProvider>
              <ModalsProvider>
                <GlobalStyles />
                <div className="grafana-app">
                  <Router history={locationService.getHistory()}>
                    {newNavigationEnabled ? <NavBarNext /> : <NavBar />}
                    <main className="main-view">
                      {pageBanners.map((Banner, index) => (
                        <Banner key={index.toString()} />
                      ))}

                      <AngularRoot ref={this.container} />
                      <AppNotificationList />
                      <SearchWrapper />
                      {this.state.ngInjector && this.renderRoutes()}
                      {bodyRenderHooks.map((Hook, index) => (
                        <Hook key={index.toString()} />
                      ))}
                    </main>
                  </Router>
                </div>
                <LiveConnectionWarning />
                <ModalRoot />
              </ModalsProvider>
            </ThemeProvider>
          </ConfigContext.Provider>
        </ErrorBoundaryAlert>
      </Provider>
```

Reply:
@fabio-silva I missed it was not a functional component. Sorry about that! Current solution is ok!!